utilgo/tlsmanager/tlsmanager.go

```diff
@@ -173,9 +174,18 @@ func (m *tlsManager) WatchServerTLSConf(serverName ClientName, handler func(conf
 		case kclient.SecretChangeTypeAdd:
 			fallthrough
 		case kclient.SecretChangeTypeUpdate:
+			if m.tlsClientCertPEM[serverName] != nil && bytes.Compare(secret.Data[TLSSecretCrtName], m.tlsClientCertPEM[serverName]) == 0 {
+				return
+			}
+
+			m.tlsClientCertPEM[serverName] = secret.Data[TLSSecretCrtName]
 			config, err = m.generateTLSConfFromKeyPair(secret.Data[TLSSecretCrtName], secret.Data[TLSSecretKeyName])
+			m.tlsConf = config
 			insecure = false
 		case kclient.SecretChangeTypeDelete:
+			if m.tlsConf == nil {
```

Comment:
Should m.tlsConf be nil in this case? Won't the m.tlsConf get set to the self signed certificate generated below?

Reply:
Yes, the server certificate shouldn't be nil in any case. Removed those lines.